link-grammar/dict-file/dictionary.c

```diff
@@ -154,6 +154,11 @@ dictionary_six_str(const char * lang,
 	}
 
 	/* Read dictionary from the input string. */
+
+	/* Make sure "." is used in string conversion to floating point.
+	 * FIXME: Use a locale-agnostic conversion function. */
+	setlocale(LC_NUMERIC, "C");
+
 	dict->input = input;
 	dict->pin = dict->input;
 	if (!read_dictionary(dict))
```

Comment:
It would be best to put the `setlocale` into `dictionary_setup_locale` which already does that kind of work.

The call to `dictionary_setup_locale` is immediately below, after the dictionary read (!) ... that does not seem right...

Reply:
`dictionary_setup_locale` is after the dictionary read because it needs to know the dictionary locale in order to set it.